vendor/k8s.io/klog/klog.go

```diff
@@ -771,24 +785,38 @@ func (l *loggingT) output(s severity, buf *buffer, file string, line int, alsoTo
 		if alsoToStderr || l.alsoToStderr || s >= l.stderrThreshold.get() {
 			os.Stderr.Write(data)
 		}
-		if l.file[s] == nil {
-			if err := l.createFiles(s); err != nil {
-				os.Stderr.Write(data) // Make sure the message appears somewhere.
-				l.exit(err)
+
+		if logging.logFile != "" {
+			// Since we are using a single log file, all of the items in l.file array
+			// will point to the same file, so just use one of them to write data.
```

Comment:
> Since we are using a single log file, all of the items in l.file array
> will point to the same file, so just use one of them to write data.

is that accurate if `SetOutputBySeverity` was called?

Reply:
@liggitt i should probably add godoc comments that when using SetOutput* they should avoid setting any command line params.

Reply:
if CLI flags and SetOutputBySeverity are mutually exclusive, and kube doesn't use SetOutputBySeverity, that's probably ok. Definitely needs documenting